scalacheck/src/main/scala/pathy/scalacheck/PathOf.scala

```diff
@@ -54,19 +54,14 @@ object PathOf {
     genRelFile[A] map (rootDir </> _)
 
   private def genRelFile[A: Arbitrary: Show]: Gen[RelFile[Sandboxed]] =
-    for {
-      d <- genRelDir[A]
-      s <- genSegment[A]
-    } yield d </> file(s)
+    sizeDistributed(genRelDir[A], genSegment[A])((d, s) => d </> file(s))
 
   private def genAbsDir[A: Arbitrary: Show]: Gen[AbsDir[Sandboxed]] =
     genRelDir[A] map (rootDir </> _)
 
   private def genRelDir[A: Arbitrary: Show]: Gen[RelDir[Sandboxed]] =
```

Comment:
Not sure what you mean by optimal. With a size of 10, the generator can produce a path with as many as 5 segments, where each dir/filename is a single char, since the `/`s are effectively counted.